packages/@aws-cdk/aws-applicationautoscaling/lib/step-scaling-policy.ts

```diff
@@ -183,7 +183,7 @@ export interface ScalingInterval {
 }
 
 function aggregationTypeFromMetric(metric: cloudwatch.IMetric): MetricAggregationType {
-  const statistic = metric.toAlarmConfig().statistic;
+  const statistic = metric.toMetricConfig().metricStat?.statistic;
```

Comment:
Don't think this is correct. Step scaling will support metric math, but `statistic` here will be `undefined` and the `case` below will throw. 

It will be safest to return `undefined` in that case, I think.